AggregateMetrics/AggregateMetrics/AzureWebApp/PerformanceCollectorModule.cs

```diff
@@ -21,14 +24,40 @@ public class PerformanceCollectorModule : ITelemetryModule
                                                                 @"\Processor(_Total)\% Processor Time"
                                                             };
 
-        public IList<PerformanceCounterCollectionRequest> Counters { get; private set; }
+        public List<PerformanceCounterCollectionRequest> Counters { get; private set; }
 
+        /// <summary>
+        /// Initializes the default performance counters.
+        /// </summary>
         public void Initialize(TelemetryConfiguration configuration)
         {
             CounterFactory factory = new CounterFactory();
 
             foreach (string counter in defaultCounters)
-                factory.GetCounter(counter);
+            {
+                try
+                {
+                    ICounterValue c = factory.GetCounter(counter);
+                    configuration.RegisterCounter(c);
+                }
+                catch (ArgumentException e)
+                {
+                    throw e;
+                }
+            }
+
+            foreach (var counter in Counters)
+            {
+                try
+                {
+                    ICounterValue c = factory.GetCounter(counter.PerformanceCounter);
+                    configuration.RegisterCounter(c);
+                }
+                catch (ArgumentException e)
+                {
+                    throw e;
```

Comment:
Same here, don't catch and throw for no purpose.  It is okay to catch, add value, and then re-throw.  "Add value" could be logging and "throw;" to preserve the stack trace or adding additional information and throwing a new wrapping exception.